### PR TITLE
fix: allows deleted documents to be stored again

### PR DESCRIPTION
**Pull request by grace0925**
Signed-off-by: Grace Liu <grace.liu@securekey.com>
Closes: #81

### CODE REVIEW

**Comment by grace0925** [comment]
When a document is deleted, couchDB does a soft delete and marks it at 'deleted'. If we attempt to fetch the deleted document with its ID, an error "Not Found: deleted" is thrown.

Currently, we do not take this into account when we check for errors from the getRevID() call which doesn't allow deleted docs to be recreated.

File: pkg/storage/couchdb/errors.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ const (
 	// Error messages returned from Kivik CouchDB client that we directly check for
 	duplicateDBErrMsgFromKivik = "Precondition Failed: The database could not be created, the file already exists."
 	docNotFoundErrMsgFromKivik = "Not Found: missing"
+	docDeletedErrMsgFromKivik  = "Not Found: deleted"
```
Comment (sandrask) [inline pkg/storage/couchdb/errors.go:42]: Do you have unit-test for this scenario "delete followed by query"?

Comment (grace0925) [reply pkg/storage/couchdb/errors.go:42]: I'll add one right now

File: pkg/storage/couchdb/couchdbstore_test.go [hunk 2]
```diff
@@ -218,6 +218,20 @@ func TestCouchDBStore_Put(t *testing.T) {
 		err := store.Put(testDocKey, []byte(testNonJSONValue))
 		require.NoError(t, err)
 	})
+	t.Run("Success: put after delete", func(t *testing.T) {
```
Comment (grace0925) [inline pkg/storage/couchdb/couchdbstore_test.go:221]: @sandrask is this what you meant?

Comment (sandrask) [reply pkg/storage/couchdb/couchdbstore_test.go:221]: This is a good test - it tests your new error.

**Review by sandrask — APPROVED** [review]

**Review by DRK3 — APPROVED** [review]
Can you also add this fix to the Aries-framework-go implementation?

**Comment by grace0925** [comment]
> Can you also add this fix to the Aries-framework-go implementation?

yep

**Comment by DRK3** [comment]
> > Can you also add this fix to the Aries-framework-go implementation?
> 
> yep

(after talking with @grace0925 offline, it turns out that the Aries version won't suffer from the same bug)

**Pull request merged**